article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoriesService.java

```diff
@@ -0,0 +1,60 @@
+package co.inventorsoft.academy.spring.service;
+
+import com.google.gson.Gson;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Data
+@RequiredArgsConstructor
+@Service
+public class CategoriesService {
+
+    private final Gson gson;
+
+    public void createCategoriesFile(List<List<String>> splitWords) throws IOException {
```

Comment:
- catch exception,
- the method is not SRP, it creates file and executed logic by counting words and fetching categories, 

For articles and users files you have repositories, but here for categories, no, it is saved in the service class, if you decide to go with the repository layer, keep it consistent.

I would write it like:

```
public void saveCategories... {
   Set<String> categories = fetchCategories(words);
   categoryRepository.save(categories);
}
```